### DISCUSSION TITLE
filepreview.el discussion

### DISCUSSION THREAD

**Discussion by karthink** [minad/consult#864] [Ideas]
Hi Daniel,

The filepreview.el idea you mention in 65dcacd7 seems good to me, are you interested in developing it further? It can help avoid duplication of effort between consult, dired-preview, (possibly) dirvish and other packages.

I was looking into adding media support to dired-preview next, and considering if it should be a separate library. Specifically, the `filepreview-handle-media` functionality from your plan is [already implemented in dirvish](https://github.com/alexluigit/dirvish/blob/main/dirvish-widgets.el#L498), and it handles most file types, generates thumbnails, does prefetching asynchronously, etc.

CC @protesilaos

**Comment by minad** [comment]
Hi Karthik,

I talked with Prot a while ago about a shared filepreview library. It
would be nice to have, for exactly the motivation you mentioned,
avoiding duplication between consult, dired-preview, dirvish, etc.
Furthermore it is architecturally cleaner if we come up with a
relatively generic and reusable API. I don't think we can use the
dirvish code currently because of the copyright assignment, but we could
ask. I'd like to develop something along these lines, but it is not
clear when I find time for it. I thought I should mentioned it again at
least given your recent enhancements to dired-preview, which I've
adopted here too in some form.
